databases/week3/mealsharing.sql

```diff
@@ -0,0 +1,34 @@
+CREATE DATABASE mealsharing DEFAULT CHARACTER SET = 'utf8mb4';
+use mealsharing;
+CREATE TABLE `Reservation` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `number_of_guests` INT(10) NOT NULL,
+  `created_date` DATETIME NOT NULL,
+  `contact_phonenumber` VARCHAR(10),
+  `contact_name` VARCHAR(50) NOT NULL,
+  `contact_email` VARCHAR(50),
+  `meal_id` int(10) unsigned NOT NULL,
+  PRIMARY KEY (`id`),
+  CONSTRAINT `fk_Meal` FOREIGN KEY (`meal_id`) REFERENCES `Meal` (`id`) ON DELETE CASCADE
```

Comment:
Remember that you need to create the Meal table first, otherwise when you run this it crashes.

Also, if we delete all reservations for a given meal, the meal entry will also be deleted, is that what you want? (It is ok, just checking )
And what happens if you update the id of the meal?

Reply:
Hi Marta, thanks for the feedback.
I will create Meal table first. Whatever order i put my tables i run only those which doesn't have foreign key and i run one by one table.
I deleted all reservations for meal id 4, still meal id 4 exists in Meal table.
And the last one i am unable to update the id of the meal, it gives error like this, Cannot delete or update a parent row: a foreign key constraint fails.  
Expecting your suggestions

Reply:
"I deleted all reservations for meal id 4, still meal id 4 exists in Meal table." <--- yes, it works the other way around: if you delete meal with id 4, all reservations for meal id 4 are also deleted ;)

"And the last one i am unable to update the id of the meal, it gives error like this, Cannot delete or update a parent row: a foreign key constraint fails." <--- this is usually because you're trying to insert a reservation or review for a meal_id that doesn't exist in the table meal.